File: Niveau 4/3 – Structures de données et Balayages/08 - Maisons pour philatélistes.py
import sys

def main():
  nbMaisons= int(input())
  maisons = list(map(int, input().split()))

  maisons.sort()
  ans = sys.maxsize

  for i in range(nbMaisons - 1):
     d = abs(maisons[i] - maisons[i + 1])
     ans = min(ans, d)
   
  print(ans)

main()

File: Niveau 4/3 – Structures de données et Balayages/08 - Maisons pour philatélistes.cpp
#include <algorithm>
#include <iostream>
#include <limits.h>
#include <vector>

using namespace std;

int main()
{

    int nbMaisons;
    cin >> nbMaisons;

    vector<int> maisons(nbMaisons, 0);

    for (int i = 0; i < nbMaisons; i++)
    {
        cin >> maisons[i];
    }

    sort(maisons.begin(), maisons.end());

    int ans = INT_MAX;
    for (int i = 1; i < nbMaisons; i++)
    {
        int d = abs(maisons[i - 1] - maisons[i]);
        ans = min(ans, d);
    }

    cout << ans;
}